### PR TITLE
Getting Started with deep neural systems in Matlab

### LINKED ISSUE #1963: How to create deep neural network in MATLAB
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Introduction:
Neural network form the base of deep learning and a subfield of machine learning where algorithm are inspired by the structure of human brain. It takes in data train themselves to recognize the pattern in this data and then predict the output for a new set of similar data.
MATLAB provides a deep learning toolbox for implementing the deep neural network. MATLAB includes the digit dataset so you don't need to download it from an external source.
In this article we will look at how to prepare your digit dataset,the structure of convolution neural network. We will also look at the Matlab code for training and validation and code for discrete testing.

### Key takeaways: 
- How to create deep neural network such as convolution neural network (CNN).
- How to train the CNN on your own set of images.
- How to recognize objects

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by jumapaul**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1963

### CODE REVIEW

File: articles/deep-neural-system-matlab/index.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
+### How to create a deep neural network using MATLAB.
+### Introduction.
+Neural networks form the base of deep learning and a subfield of machine learning where algorithms are inspired by the structure of the human brain. It takes in data train themselves to recognize the pattern in this data and then predict the output for a new set of similar data.
```
Comment (paulodhiambo) [inline articles/deep-neural-system-matlab/index.md:3]: Consider changing "Neural networks form the base of deep learning and a subfield of machine learning..." to "Neural networks form the base of deep learning, a subfield of machine learning..." . "It takes in data train themselves to recognize the pattern in this data..." to "It takes in data, train itself to recognize the pattern in this data..."

File: articles/deep-neural-system-matlab/index.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+MATLAB provides a deep learning toolbox for implementing the deep neural network. MATLAB includes the digit dataset so you don't need to download it from an external source.
+In this article, we will look at how to prepare your digit dataset, the structure of a convolution neural network. We will also look at the Matlab code for training and validation and code for discrete testing.
+
+### Prerequisite.
+1. Knowledge in Matlab.
+2. Matlab installed on your computer.
```
Comment (paulodhiambo) [inline articles/deep-neural-system-matlab/index.md:9]: Consider adding links to Matlab download site. Reference a getting started article that is already published [here](https://www.section.io/engineering-education/getting-started-with-matlab/).

File: articles/deep-neural-system-matlab/index.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+### MATLAB’S digit dataset.
+Matlab itself includes the digit dataset. The digit dataset consists of 10,000 grayscale images of handwritten digits and each digit has 1000 images of the respective folders. The images are of size 28 by 28 pixels and they have associated labels.
+
+### How to locate the image dataset.
+- Go to the installation folder of Matlab and then to the toolbox.
+- Click the nnet and then choose the nndemos folder
+- Select the mdataset and then the digit dataset folder.
+
+> Note that after locating the dataset folder, you can copy it to your working directory.
```
Comment (paulodhiambo) [inline articles/deep-neural-system-matlab/index.md:21]: Provide a screenshot showing the full path

File: articles/deep-neural-system-matlab/index.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+trainingOption(solverName,’name’,’value’)
+```
+In the training option, we have to define the solver name.
+Example
+```Matlab
+trainingOption(‘sgdm’,’learnRateSchedule’,’piecewise’,’learnRateDropFactor’,0.2,
+learnRateDropPeriod’,5,’maxEpochs’,20,’miniBatchSize’,64,’plots’,’trainingProgress’)
+```
+We use the plot factor here to see the progress of the training in a graph. The graph will be of output against epochs.
+There are various parameters and if you want to learn all, check the documentation.
```
Comment (paulodhiambo) [inline articles/deep-neural-system-matlab/index.md:112]: provide a link to documentation

**Review by paulodhiambo — CHANGES_REQUESTED** [review]
@jumapaul amazing tutorial. Revise the changes and ping me when done

**Comment by jumapaul** [comment]
x-special/nautilus-clipboard
copy
file:///home/pauljuma/Documents/MATLAB/articles/image1.jpg
file:///home/pauljuma/Documents/MATLAB/articles/image2.jpg
file:///home/pauljuma/Documents/MATLAB/articles/image3.jpg
file:///home/pauljuma/Documents/MATLAB/articles/image4.jpg
file:///home/pauljuma/Documents/MATLAB/articles/image5.jpg
file:///home/pauljuma/Documents/MATLAB/articles/image6.jpg
file:///home/pauljuma/Documents/MATLAB/articles/image7.png
file:///home/pauljuma/Documents/MATLAB/articles/image8.png

**Comment by jumapaul** [comment]
@paulodhiambo changes made

**Comment by hectorkambow** [comment]
[Getting Started with deep neural systems in Matlab #2063.pdf](https://github.com/section-engineering-education/engineering-education/files/6379137/Getting.Started.with.deep.neural.systems.in.Matlab.2063.pdf)

Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

@jumapaul

**Comment by paulodhiambo** [comment]
@jumapaul amazing article. @hectorkambow you can it a final review.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
@jumapaul Overall seems like a very helpful tutorial. 

Please go over my final edits closely. As I found this article really difficult to get through and some sentences were difficult to follow along. 

I had half a mind to close the PR - please be sure to only include fully polished article (ready to be published as is). 

As we do not have the resources to review and edit articles - next time we will simply close the PR. 👍  

thank you

cc @paulodhiambo please review the edits as well - as articles like this should not be pushed for final review 🚀  
thank you

**Pull request merged**